app/models/user.rb

```diff
@@ -16,9 +18,9 @@ class User < ActiveRecord::Base
   attr_accessible *(ACCESSIBLE_ATTRIBUTES + [:admin]), :as => :admin
 
   validates_presence_of :username
-  validates_uniqueness_of :username
+  validates :username, uniqueness: { case_sensitive: false }
   validates_format_of :username, :with => /\A[a-zA-Z0-9_-]{3,15}\Z/, :message => "can only contain letters, numbers, underscores, and dashes, and must be between 3 and 15 characters in length."
-  validates_inclusion_of :invitation_code, :on => :create, :in => INVITATION_CODES, :message => "is not valid", if: ->{ User.using_invitation_code? }
+  validates_inclusion_of :invitation_code, :on => :create, :in => INVITATION_CODES, :message => "is not valid", if: -> { !requires_no_invitation_code? && User.using_invitation_code? }
```

Comment:
Looks great to me! Awesome work!
On Thu, Mar 17, 2016 at 8:36 AM Andrew Cantino <cantino@gmail.com> wrote:

> I don't feel strongly. It confused me for a moment when compared to
> User.using_invitation_code? But it's not hard to figure out :)
>
> On Thu, Mar 17, 2016 at 2:11 AM Dominik Sander <notifications@github.com>
> wrote:
>
>> In app/models/user.rb
>> <https://github.com/cantino/huginn/pull/1330#discussion_r56473845>:
>>
>> >    validates_format_of :username, :with => /\A[a-zA-Z0-9_-]{3,15}\Z/, :message => "can only contain letters, numbers, underscores, and dashes, and must be between 3 and 15 characters in length."
>> > -  validates_inclusion_of :invitation_code, :on => :create, :in => INVITATION_CODES, :message => "is not valid", if: ->{ User.using_invitation_code? }
>> > +  validates_inclusion_of :invitation_code, :on => :create, :in => INVITATION_CODES, :message => "is not valid", if: -> { !requires_no_invitation_code? && User.using_invitation_code? }
>>
>> user.this_user_requires_no_invitation_code? or in this case
>> self.this_user_requires_no_invitation_code? sound a bit redundant to me.
>>
>> —
>> You are receiving this because you were mentioned.
>>
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cantino/huginn/pull/1330/files/df349a73028f623ba9e0dbf68ac6736e1134f2a4#r56473845>
>>
>